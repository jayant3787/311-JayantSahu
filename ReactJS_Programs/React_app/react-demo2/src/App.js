
import { Route, Routes } from "react-router";
// import ReduxDemo from './components/ReduxDemo';
import ReduxDemoContainer from "./components/ReduxDemoContainer";
import store from "./store/myStore";
import { Provider } from 'react-redux';


import './App.css';
import NavBar from "./components/NavBar";
import NewsRenderDemo from "./components/NewsRenderDemo";

function App() {
  return (
    <Provider store={store}>
    <Routes>
    <Route path="/news" element={<NewsRenderDemo />} />
    <Route path="/redux/redux-demo2" element={<ReduxDemoContainer />} />
    <Route path="/" element={<Home />} />
    </Routes>
    </Provider>


    
  );
};

function Home() {
  return (
    <NavBar />

   
  )
}


export default App;
