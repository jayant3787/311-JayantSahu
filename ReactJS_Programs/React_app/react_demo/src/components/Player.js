import { Component } from "react";
// import Message from "./scoreboard/Message";
class Player extends Component{
  constructor(props){
    super(props);

    this.state={
      lastScore:0,
      score:props.score
    }
  }


  f1 =()=>{
    let tempScore = Math.floor(Math.random()*10)
    this.setState({score:this.state.score+tempScore,lastScore:tempScore})

  };
    render(){
        return(
<div style={{border:"5px solid orange"}}>
        <h2>
          Player Name: {this.props.name}<br />
          <button onClick={this.props.someFunc}>Click here to go to the parent</button>
        </h2>
        <h3>Score: {this.state.score} <br /></h3>
        {/* <Message /> */}
      </div>
        )
    }



}

export default Player;