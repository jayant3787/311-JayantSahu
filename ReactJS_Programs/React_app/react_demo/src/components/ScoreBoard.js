import { Component } from "react";

class ScoreBoard extends Component {

    changeScores = () =>{
        
    }


 
  render() {
    return (
      <div style={{border:"5px solid orange"}}>
        {/* <img src={this.props.image} alt="Profile img " /> */}
        <h1>
         **********SCOREBOARD**********
         {this.props.children}

         <button onClick={this.changeScores}>
            click to increase the scores
          </button>
        </h1>
      </div>
    );
  }
}

export default ScoreBoard;
