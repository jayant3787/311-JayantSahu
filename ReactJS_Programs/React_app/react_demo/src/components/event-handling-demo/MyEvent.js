import { useState } from "react";
const MyEvent = () => {

    function handleSubmit(e) {
        e.preventDefault();
        alert('type of event = '+e.type);
        alert('target of event = '+e.target.name);



        //check is in the form
        let age = document.forms[0].age.value;
        console.log('You clicked submit.');

        if (age > 60) {
            console.log("if age is > 60 submit the form");
            document.forms[0].submit();
        };
    }

    const handleEvent2 = (e, param2) =>{
        e.preventDefault();
        alert('handleEvent2 type of event : '+e.type);
        alert('handleEvent2 target of event : '+e.target.name);
        alert("handleEvent2 target of event : "+param2)


    }
    const [age,setAge] = useState(0);

    const handleEvent3 = (e) => {
        e.preventDefault();
        setAge(25);
    }

    return (
        <div style={{border: "4px solid blue"}}>
        age : {age}
            <form name="form1" onSubmit={handleSubmit} action="backend.php">
                <p>Events Demo</p>
                <input type="text" name="age" />
                <button id="btn1" type="submit" name="button1">Submit</button><br />

                <button type="submit" name="button2" 
                onClick={(e) => handleEvent2(e,'extra.info')}>Submit button 2
                </button><br/>
               
            </form>
            <button id="btn3" type="submit" name="button3" 
                onClick={handleEvent3}>Submit button 3
                </button><br/><br/>
        </div>

    );

}
export default MyEvent;