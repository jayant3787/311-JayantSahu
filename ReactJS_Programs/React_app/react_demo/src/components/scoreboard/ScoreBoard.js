import React, { Component } from "react";
import Player from "../Player";
import Message from "./Message";

class ScoreBoard extends Component {
  constructor(props) {
    super(props);
     this.ref1 = React.createRef();
     this.ref2 = React.createRef();
     this.ref3 = React.createRef();

  }

  // demo of showcasing calling child's function from this parent
  changeScores = () => {
    this.ref1.current.f1();
    this.ref2.current.f1();
    // alert(this.ref1.current.state.score);
    // alert(this.ref2.current.state.score);
    var score1 = this.ref1.current.state.score;
    var score2 = this.ref2.current.state.score;
    if(score1>50){
      // alert(score1);
      this.ref3.current.setState((state,props)=>({message:"Winner is Virat"}))
    }
    else if(score2>50){
      // alert(score1);
      this.ref3.current.setState((state,props)=>({message:"Winner is Rohit"}))

    }


  };

  f2=()=>{
    alert("in scoreboard parent");
  }

  render() {
    return (
      <div style={{ border: "5px solid orange" }}>
        <h1>
          **********SCOREBOARD**********
          <br />
          INDIA</h1>
          <Player name="Virat" ref={this.ref1} score={0} someFunc={this.f2} />
          <Player name="Rohit" ref={this.ref2} score={0} someFunc={this.f2}/>
          <Message ref={this.ref3}/>
          {/* {this.props.children} */}
          <button onClick={this.changeScores}>
            CLICK HERE TO CHANGE THE SCORES RANDOMLY
          </button>
        
      </div>
    );
  }
}

export default ScoreBoard;
