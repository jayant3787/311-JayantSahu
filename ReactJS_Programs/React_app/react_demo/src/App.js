// import logo from './logo.svg';
import "./App.css";
import Person from "./components/Person";
import User from "./components/User";
import User2 from "./components/User2";

import Address from "./components/Address";
import Hobby from "./components/Hobby";
import Qualification from "./components/Qualification"
import ScoreBoard from "./components/scoreboard/ScoreBoard";
import LifeCycleDemo from "./components/lifecycle_demo/Demo";
import Fruits from "./components/context-demo/Fruit";
import FruitContext from "./components/context-demo/FruitContext";
import Greeting from "./components/functionalComponent_demo/Greeting";
import GreetingES6 from "./components/functionalComponent_demo/GreetingES6";
import PersonComponent from "./components/functionalComponent_demo/PersonComponent";
import ScoreBoardComponent from "./components/functionalComponent_demo/ScoreboardComponent";
// import Player from "./components/Player";

function App() {
  return (
    
    <FruitContext.Provider value={{x:2,y:5}}>
    {/* functional coponent */}
    <Greeting name="Jayant Sahu"/>
    <GreetingES6 name="JAYANT"/>
    <PersonComponent name="Virat Kohli"/>
    <ScoreBoardComponent />

    <div className="App">
      <header className="App-header">
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        Hello JAYANT
        {/* <p>
          Edit <code>src/App.js</code> and save to reload.
        </p> */}
        <br />
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Click here to Learn React
        </a>
        <User name="Jayant" />
        <User name="Meghal" />
        <Person
          name="Jayant Sahu"
          popularity={1}
          image="images/logo192.png"
          fName="Santosh"
          mName="Uma"
          email="abc@gmail.com"
        />
        <ScoreBoard  >
          {/* <Player name="Virat" score={60} /> */}
          {/* <Player name="Rohit"  score={80}/> */}
        </ScoreBoard>
        <LifeCycleDemo name="Jay"/>
        <Fruits />

        {/* //user 2 component is coded such a way that it will render all the children component 
            //therefore we can pass any number(even zero) of hobby,qualification,address
            //components as its children  */}
        <User2 name="Uma Sahu">
        <Address address= {["2943 sangam chowk","101 ashoka ratna ", "102 Laxmi nagar"]} />
        <Hobby hobby={["Music , Cricket "]}/>
        <Qualification qualification={["BCA","MCA"]} />

        </User2>

        {/* //inside this user2 we are only passing address, hobby
            // children */}

        <User2 name="Santosh Sahu">
        <Address address= {["2943 sangam chowk","101 ashoka ratna ", "102 Laxmi nagar"]} />
        <Hobby hobby={["Music , Cricket, Netflix"]}/>

        </User2>
      </header>
    </div>
    </FruitContext.Provider>

  );
}

export default App;
