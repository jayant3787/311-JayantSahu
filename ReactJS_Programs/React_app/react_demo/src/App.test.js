import App from './App';
import React from 'react';
// import ShallowRenderer from 'react-test-renderer/shallow'
import { mount, shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ScoreBoard from "./components/scoreboard/ScoreBoard";
import HelloWorld from './PureComp-demo';
import toJson from 'enzyme-to-json';
import Player from './components/Player';
// import sinon from 'sinon';
import MyEvent from './components/event-handling-demo/MyEvent';


configure({ adapter: new Adapter() });


// const renderer = new ShallowRenderer();

test('two plus two is four', () => {
  expect(2 + 2).toBe(4);
});

// doing the grouping putting 2 test cases into 1
describe('basic app rendering test cases', () => {
  test("check app comes without any issues", () => {
    shallow(<App />);
  });

  test("app renders the text inside it", () => {
    const wrapper = mount(<App />);
  });
});


test('shallow app rendering should not display child content', () => {
  var wrapper = shallow(<App />);
  const someContent = "CHANGE THE HOBBY";
  expect(wrapper.contains(someContent)).toEqual(false);
});



describe('HTML content rendering test suit', () => {
  test('app renders the HTML content inside it', () => {
    var wrapper = shallow(<App />);
    const someContent = <h4>Click here to Learn React</h4>;
    expect(wrapper.contains(someContent)).toEqual(true);
  });

  test('app render the HTML content inside it', () => {
    var wrapper = shallow(<App />);
    const someContent = <h4>Click here to Learn Reacter</h4>;
    expect(wrapper.contains(someContent)).toEqual(false);
  });

  test('app has a 2 h4 element', () => {
    var wrapper = shallow(<App />);
    // use CSS selector to find all h4 element
    const noOfh4Elements = wrapper.find('h4').length; // this will return an array
    expect(noOfh4Elements).toEqual(2);
  });

  test('app has a single ScoreBoard component', () => {
    var wrapper = mount(<App />);
    const isScoreBoard = wrapper.find('ScoreBoard').length;
    expect(isScoreBoard).toEqual(1); //OR
    // expect(isScoreBoard).to.have.lengthOf(1);
  });

  test('app has a single ScoreBoard component and has content....', () => {
    // var wrapper = mount(<App />);
    const wrapper1 = mount(<ScoreBoard />);
    expect((wrapper1.text()).indexOf("INDIA") !== -1).toEqual(true);//OR

    // the below way is also used
    // expect(wrapper1.contains("INDIA")).toEqual(true);

  });

  test('score board is same or not', () => {
    // var wrapper = mount(<App />);
    const tree = mount(<HelloWorld />);
    expect(toJson(tree)).toMatchSnapshot();

  });
});



describe('h2 element rendering test suit', () => {
  test('app renders find the element by id', () => {
    var wrapper = mount(<App />);
    const findH2 = wrapper.find('#myH2');
    expect(findH2.contains("HI")).toEqual(true);
  });

  test('app renders find the element by id and display its content', () => {
    var wrapper = mount(<App />);
    expect((wrapper.text()).indexOf("HI") !== -1).toEqual(true);
  });

  // i am passing the props hence keeping true to pass
  test('app renders props are passed or not', () => {
    var wrapper = mount(<Player name="jayu" />);
    expect(wrapper.contains("jayu")).toEqual(true);
  });

  //not passing the props hence keeping false to pass
  test('app renders props are passed', () => {
    var wrapper = mount(<Player />);
    expect(wrapper.contains("jayu")).toEqual(false);
  });
});


test('simulates submit events', () => {
  // const onButtonSubmit = sinon.spy();
  const wrapper = shallow(<MyEvent />);
  wrapper.find('#btn3').simulate('click');
  console.log("simulate test case"+wrapper.text());
  expect(wrapper.contains("25")).toEqual(true);
});




