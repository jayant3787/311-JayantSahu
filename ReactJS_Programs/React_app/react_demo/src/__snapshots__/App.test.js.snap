// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML content rendering test suit score board is same or not 1`] = `
<HelloWorld>
  <p>
    Aloha
  </p>
</HelloWorld>
`;
